mediaengine.go

```diff
@@ -432,3 +293,44 @@ func (m *MediaEngine) collectStats(collector *statsReportCollector) {
 		collector.Collect(stats.ID, stats)
 	}
 }
+
+// TODO Sean-Der, everything below is not finished
+
+// Update the MediaEngine from a remote description
+func (m *MediaEngine) updateFromRemoteDescription(desc sdp.SessionDescription) error {
+	for _, media := range desc.MediaDescriptions {
+		switch {
+		case !m.negotiatedAudio && strings.EqualFold(media.MediaName.Media, "audio"):
+			m.negotiatedAudio = true
+		case !m.negotiatedVideo && strings.EqualFold(media.MediaName.Media, "video"):
+			m.negotiatedVideo = true
+		default:
+			continue
+		}
+
+		// * convert media.Attributes in []RTPCodecParameters
+		// * Update media.videoCodecs (or media.audioCodecs)
+		//   - set negotiated=true
+		//   - set the proper payloadtype
+		//   - set the proper fmtp
+	}
+	return nil
+}
+
+// negotiatedCodecsForType and negotiatedCodecsforType both return a properly
+// filtered view of the MediaEngine.
+// Note if negotiation hasn't taken place yet all values are assumed enabled
+
+func (m *MediaEngine) negotiatedCodecsForType(typ RTPCodecType) []RTPCodecParameters {
```

Comment:
I think this method should be removed, because if MediaEngine can be reused, then it can't store PeerConnection specific information such as negatiation status and PeerConnection specific payload types. Instead updateFromRemoteDescription could return an RTPParameters object, and it could be stored for the peerconnection and passed to RTPSender/etc. when needed.

Reply:
This doesn't matter if MediaEngine is not shared.